protoc-gen-openapiv2/internal/genopenapi/template_test.go

```diff
@@ -3905,118 +4053,83 @@ func TestSchemaOfField(t *testing.T) {
 			refs: make(refMap),
 			expected: openapiSchemaObject{
 				schemaCore: schemaCore{
-					Type:   "integer",
-					Format: "int32",
-				},
-				Title:       "field title",
-				Description: "field description",
-			},
-		},
-		{
-			field: &descriptor.Field{
-				FieldDescriptorProto: &descriptorpb.FieldDescriptorProto{
-					Name:     proto.String("message_field_option"),
-					Label:    descriptorpb.FieldDescriptorProto_LABEL_OPTIONAL.Enum(),
-					Type:     descriptorpb.FieldDescriptorProto_TYPE_MESSAGE.Enum(),
-					TypeName: proto.String(".example.Empty"),
-				},
-			},
-			openAPIOptions: &openapiconfig.OpenAPIOptions{
-				Field: []*openapiconfig.OpenAPIFieldOption{
-					{
-						Field:  "example.Message.message_field_option",
-						Option: jsonSchema,
+					Type: "array",
+					Items: &openapiItemsObject{
+						schemaCore: schemaCore{
+							Type:   "string",
+							Format: "int64",
+						},
+						MultipleOf:       100,
+						Maximum:          101,
+						ExclusiveMaximum: true,
+						Minimum:          1,
+						ExclusiveMinimum: true,
+						MaxLength:        10,
+						MinLength:        3,
+						Pattern:          "[a-z]+",
+						UniqueItems:      true,
+						MaxProperties:    33,
+						MinProperties:    22,
+						Required:         []string{"req"},
+						ReadOnly:         true,
 					},
 				},
-			},
-			refs: refMap{".example.Empty": struct{}{}},
-			expected: openapiSchemaObject{
-				schemaCore: schemaCore{
-					Ref: "#/definitions/exampleEmpty",
-				},
 				Title:       "field title",
 				Description: "field description",
-			},
-		},
-		{
-			field: &descriptor.Field{
-				FieldDescriptorProto: &descriptorpb.FieldDescriptorProto{
-					Name:    proto.String("required_via_field_behavior_field"),
-					Type:    descriptorpb.FieldDescriptorProto_TYPE_STRING.Enum(),
-					Options: requiredFieldOptions,
-				},
-			},
-			refs: make(refMap),
-			expected: openapiSchemaObject{
-				schemaCore: schemaCore{
-					Type: "string",
-				},
-				Required: []string{"required_via_field_behavior_field"},
-			},
-		},
-		{
-			field: &descriptor.Field{
-				FieldDescriptorProto: &descriptorpb.FieldDescriptorProto{
-					Name:    proto.String("readonly_via_field_behavior_field"),
-					Type:    descriptorpb.FieldDescriptorProto_TYPE_STRING.Enum(),
-					Options: outputOnlyOptions,
-				},
-			},
-			refs: make(refMap),
-			expected: openapiSchemaObject{
-				schemaCore: schemaCore{
-					Type: "string",
-				},
-				ReadOnly: true,
-			},
-		},
-		{
-			field: &descriptor.Field{
-				FieldDescriptorProto: &descriptorpb.FieldDescriptorProto{
-					Name:     proto.String("message_field"),
-					TypeName: proto.String(".example.Message"),
-					Type:     descriptorpb.FieldDescriptorProto_TYPE_MESSAGE.Enum(),
-					Options:  requiredFieldOptions,
-				},
-			},
-			refs: refMap{".example.Message": struct{}{}},
-			expected: openapiSchemaObject{
-				schemaCore: schemaCore{
-					Ref: "#/definitions/exampleMessage",
-				},
```

Comment:
There are way too many changes in this file to make sense of - did you need to change any existing tests, and did you add a new test for this functionality? I can't tell.

Reply:
I have mistakenly commented these test case and i have only added two test cases